src/qt/guiutil.cpp

```diff
@@ -941,6 +942,38 @@ QString formatPingTime(double dPingTime)
     return (dPingTime == std::numeric_limits<int64_t>::max()/1e6 || dPingTime == 0) ? QObject::tr("N/A") : QString(QObject::tr("%1 ms")).arg(QString::number((int)(dPingTime * 1000), 10));
 }
 
+QString formatDataSizeValue(uint64_t uValue)
+{
+    // Why handle these comparisons directly, instead of a clever algorithm?
+    // This is likely to be called in a tight loop, so avoid the overhead of
+    // setting up a constant list and walking an iterator.
+    static const uint64_t TERABYTE_SIZE = UINT64_C(1024*1024*1024*1024);
+    static const uint64_t GIGABYTE_SIZE = UINT64_C(1024*1024*1024);
+    static const uint64_t MEGABYTE_SIZE = UINT64_C(1024*1024);
+    static const uint64_t KILOBYTE_SIZE = UINT64_C(1024);
+
+    QString unitFormat = QObject::tr("%1");
```

Comment:
Can we make this formatting mask "%1 B"?

Reply:
Done, and squashed everything into a single logical commit.